### PR TITLE
[ENH] Made base_api_url an environment variable

### LINKED ISSUE #35: Make base_api_url an Environment Variable
- Make the base_api_url from url_generator.py an environment variable.

### PR DESCRIPTION
**Pull request by Raya679**
- Closes #35 

Changes proposed in this pull request:

- Made base_api_url an environment variable.
- Updated documentation.

## Checklist
_This section is for the PR reviewer_

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see our [Contributing Guidelines](https://neurobagel.org/contributing/CONTRIBUTING#pull-request-guidelines) for more info)_
- [x] PR has a label for the release changelog or `skip-release` (to be applied by maintainers only)
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [x] Tests pass
- [x] Checks pass

For new features:
- [x] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

**Comment by Raya679** [comment]
Hii @rmanaem, I have made the base_api_url an environment variable here. Please take a look and suggest any changes if required.
Thanks!

File: app/api/url_generator.py [hunk 1]
```diff
@@ -22,7 +26,7 @@ def get_api_url(user_query: str) -> str:
     Returns:
         str: The constructed API URL.
     """
-    base_api_url = "https://api.neurobagel.org/query/?"
+    base_api_url = os.getenv("BASE_API_URL", "None")
```
Comment (rmanaem) [inline app/api/url_generator.py]:
```suggestion
    base_api_url = os.getenv("NB_API_QUERY_URL", "None")
```

File: README.md [hunk 2]
```diff
@@ -18,8 +18,10 @@ The aim of query-tool-ai would be to make this search process more user-friendly
   ### Set the environment variables - 
    | Environment Variable   | Type    | Required                                 | Default Value if Not Set | Example                                                   |
    | ---------------------- | ------- | ---------------------------------------- | ------------------------ | --------------------------------------------------------- |
+   | `BASE_API_URL`                 | string  | Yes                                       | `None`                | `https://api.neurobagel.org/query/?`                                               |
```
Comment (rmanaem) [inline README.md]:
```suggestion
   | `NB_API_QUERY_URL`                 | string  | Yes                                       | `None`                | `https://api.neurobagel.org/query/?`                                               |
```

File: tests/test_url_generator.py [hunk 3]
```diff
@@ -105,13 +105,14 @@ def test_get_api_url(
     mock_input_side_effect,
     expected_output,
 ):
-    with patch(
-        "app.api.url_generator.extract_information",
-        return_value=mock_llm_response,
-    ):
-        if mock_input_side_effect is not None:
-            with patch("builtins.input", side_effect=mock_input_side_effect):
+    with patch.dict("os.environ", {"BASE_API_URL": "https://api.neurobagel.org/query/?"}):
```
Comment (rmanaem) [inline tests/test_url_generator.py]:
```suggestion
    with patch.dict("os.environ", {"NB_API_QUERY_URL": "https://api.neurobagel.org/query/?"}):
```

**Review by rmanaem — CHANGES_REQUESTED** [review]
Thanks for the PR @Raya679!
I left suggestions to rename the env var to essentially mirror the env var we are already using in the query tool.
:cherries: I think it's a good idea to have a way to make sure the new env var is set, throw an error to inform the user in case it's not and a test for that. Since the implementation simply concatenates the env var with the output from the llm to send the request, we can end up with http errors if the env var is not set properly. Let me know what you think.

**Comment by Raya679** [comment]
Hi @rmanaem , thanks for the review!
I have made the changes suggested. Also I have removed the default value `None` of NB_API_QUERY_URL since now we get an error if it is not set.
Please do suggest any changes if required.

File: app/api/url_generator.py [hunk 4]
```diff
@@ -22,7 +26,10 @@ def get_api_url(user_query: str) -> str:
     Returns:
         str: The constructed API URL.
     """
-    base_api_url = "https://api.neurobagel.org/query/?"
+    base_api_url = os.getenv("NB_API_QUERY_URL")
+    if not base_api_url:
+        return "Error: The NB_API_QUERY_URL environment variable is not set. Please configure it and try again."
+
```
Comment (rmanaem) [inline app/api/url_generator.py]: I think it's better practice to raise a runtime error here similar to what we have in the api and check for that in the test:
https://github.com/neurobagel/api/blob/c7c1839d4a013c7a3d3b0567174dbf24e81c97f2/app/main.py#L92-L99

**Review by rmanaem — CHANGES_REQUESTED** [review]
:cherries:

**Comment by Raya679** [comment]
Hii @rmanaem, I have made the changes.
Please take a look and do suggest any changes if required.

Thanks!

**Review by rmanaem — APPROVED** [review]
:cook:

**Pull request merged**